rules/dead-code/src/Rector/MethodCall/RemoveEmptyMethodCallRector.php

```diff
@@ -86,66 +79,37 @@ public function refactor(Node $node): ?Node
             return null;
         }
 
-        /** @var ObjectType $type */
         $type = $scope->getType($node->var);
-
-        if ($node->var instanceof PropertyFetch) {
-            /** @var ObjectType|ThisType $type */
-            $type = $scope->getType($node->var->var);
-        }
-
-        if ($type instanceof ThisType) {
-            return null;
-        }
-
         if (! $type instanceof ObjectType) {
             return null;
         }
 
-        /** @var ClassReflection|null $classReflection */
-        $classReflection = $type->getClassReflection();
-
-        if ($classReflection === null) {
-            return null;
-        }
-
-        $className = $type->getClassName();
-        if (is_a($className, Node::class, true)) {
-            return null;
-        }
-
-        /** @var Class_|null $class */
-        $class = $this->classReflectionToAstResolver->getClass($classReflection, $className);
-
+        $class = $this->classReflectionToAstResolver->getClassFromObjectType($type);
         if ($class === null) {
             return null;
         }
 
-        if ($this->isNonEmptyMethod($class, $node)) {
+        if (! $this->isEmptyMethod($class, $node)) {
             return null;
         }
 
-        try {
-            $this->removeNode($node);
-        } catch (ShouldNotHappenException $shouldNotHappenException) {
-            return null;
-        }
+        $this->removeNode($node);
 
         return $node;
     }
 
-    private function isNonEmptyMethod(Class_ $class, MethodCall $methodCall): bool
+    private function isEmptyMethod(Class_ $class, MethodCall $methodCall): bool
     {
-        /** @var Identifier $methodIdentifier */
-        $methodIdentifier = $methodCall->name;
-        /** @var ClassMethod|null $classMethod */
-        $classMethod = $class->getMethod((string) $methodIdentifier);
```

Comment:
@samsonasik Just FYI, this actually causes a bug:

```php
(string) $methodIdentifier
```

In case property/method name is magic, there is not `Identifier` but an `Expr`.
That's why it's better to sue `$this->getName(...)` instead of `(string) $name`

Such magic calls look like these:

```php
$this->$magic()
$this->$magicProperty
```

Reply:
@TomasVotruba your change make rector-ci error: 

```bash                                                                                
 [ERROR] Could not process                                                      
         "packages/node-type-resolver/src/NodeTypeResolver/ScalarTypeResolver.ph
         p" file, due to:                                                       
         "leaveNode() returned invalid value of type integer".                                                                                                  
```

Reply:
Thanks, I plan to fix CI before merge